09-reproducible-data.Rmd

```diff
@@ -1 +1,83 @@
+
+### Parameterization
+
+Another great feature of Quarto and R Markdown is the ability to reduce repetitive code by parameterizing the files. Parameters can control various aspects of the analysis, such as dates, geography, or other analysis variables. By parameterizing our code, we can define and modify these parameters to explore different scenarios or inputs. For example, suppose we start by creating a document that provides survey analysis results for Michigan but then later decide we want to look at multiple states. In that case, we can define a `state` parameter and rerun the same analysis for other states like Wisconsin without having to edit the code throughout the document.
+
+Parameters can be defined in the header or code chunks of our Quarto or R Markdown documents and easily be modified and documented. Thus, we are reducing errors that may occur by manually editing code throughout the script, and it is a flexible way for others to replicate the analysis and explore variations.
+
+## Other Tips for Reproducibility
+
+### Random Number Seeds  
+
+Some tasks in survey analysis require randomness, such as imputation, model training, or creating random samples. By default, the random numbers that R generates will change each time we rerun the code, making it difficult to reproduce the same results. By "setting the seed," we can control the randomness and ensure that the random numbers remain consistent whenever we rerun the code. Others can use the same seed value to reproduce our random numbers and achieve the same results, facilitating reproducibility.
+
+In R, we can use the `set.seed()` function to control the randomness in our code. Set a seed value by providing an integer to the function:
+
+```r
+set.seed(999)
+
+runif(5)
+```
+
+The `runif()` function generates five random numbers from a uniform distribution. Since the seed is set to `999`, running `runif()` multiple times will always produce the same sequence:
+
+```
+[1] 0.38907138 0.58306072 0.09466569 0.85263123 0.78674676
+```
+
+The choice of the seed number is up to the researcher. For example, this could be the date (`20240102`) or time of day (`1056`) when the analysis was first conducted, a phone number (`8675309`), or the first few numbers that come to mind (`369`). As long as the seed is set for a given analysis, the actual number can be up to the researcher to decide.  However, it is important to note that `set.seed()` should be used *before* random number generation but is only necessary once per program to make the entire program reproducible. For example, we could set the seed at the top of a program where libraries are loaded.
+
+### Descriptive Names and Labels
```

Comment:
Maybe as part of this section or as its own, I think we need to mention functions for reproducibility and as a way to organize code.